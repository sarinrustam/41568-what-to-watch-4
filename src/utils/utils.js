export const FILTER_ALL_GENRES = `All Genres`;

export const PAGE_FILTERS = [`Overview`, `Details`, `Reviews`];

export const COUNT_LIMIT_MOVIES = 8;

export const CommentLength = {
  MIN: 50,
  MAX: 400,
};

export const Buttons = {
  ESC: `ESC`,
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
};
