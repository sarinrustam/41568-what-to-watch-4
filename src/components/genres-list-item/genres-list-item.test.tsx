import * as React from "react";
import * as renderer from "react-test-renderer";
import {PAGE_FILTERS} from "../../utils/utils";

import GenresListItem from "./genres-list-item";

describe(`Render GLI`, () => {
  it(`should GenresListItem render correctly`, () => {
    const handleSetCurrentGenre = jest.fn();

    const tree = renderer.create(
        <GenresListItem
          genre={PAGE_FILTERS[0]}
          isActive={true}
          onSetCurrentGenre={handleSetCurrentGenre}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
