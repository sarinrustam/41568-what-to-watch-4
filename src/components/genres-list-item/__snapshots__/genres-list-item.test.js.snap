// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GLI should GenresListItem render correctly 1`] = `
<li
  className="catalog__genres-item catalog__genres-item--active"
>
  <a
    className="catalog__genres-link"
    href="#"
    onClick={[Function]}
  >
    Overview
  </a>
</li>
`;
